### PR TITLE
Render transcript filter in SRT for genes only

### PR DESCRIPTION
**Pull request by jernestmyers**
Resolves https://github.com/VEuPathDB/web-monorepo/pull/523#issuecomment-1812597494

**NOTE**: Pulling the data to check if it's a `transcript` feels brittle. Is there a more robust way?

### CODE REVIEW

**Comment by dmfalke** [comment]
> **NOTE**: Pulling the data to check if it's a `transcript` feels brittle. Is there a more robust way?

I thought the record class object included the available view filters, but I'm not seeing that information. So, this might be the only reliable way.

File: packages/libs/web-common/src/components/reporters/SequenceFormFactory.jsx [hunk 1]
```diff
@@ -90,7 +90,11 @@ const createSequenceForm = (
       includeSubmit,
       viewFilters,
       updateViewFilters,
+      resultType,
     } = props;
+    const shouldRenderTranscriptFilter =
+      resultType?.step?.recordClassName &&
+      resultType?.step?.recordClassName === 'transcript';
```
Comment (dmfalke) [inline packages/libs/web-common/src/components/reporters/SequenceFormFactory.jsx:97]: You should use https://github.com/VEuPathDB/web-monorepo/blob/b6b240468f82ccc6fd2051b1ff2a72782efb7857/packages/libs/wdk-client/src/Utils/WdkResult.ts#L47-L48 to get the `recordClassName` of the `resultType`:

```ts
const resultTypeDetailsResult = useWdkService(wdkService => getResultTypeDetails(wdkService, resultType));
const shouldRenderTranscriptFilter = resultTypeDetailsResult.value?.recordClassName === 'transcript';
```

**Review by dmfalke — CHANGES_REQUESTED** [review]
This looks good. See my suggestion.

File: packages/libs/web-common/src/components/reporters/SequenceFormFactory.jsx (hunk 1 above)
Comment (jernestmyers) [reply packages/libs/web-common/src/components/reporters/SequenceFormFactory.jsx:97]: I'm getting an infinite loop with this approach. Any idea why?

**Review by dmfalke — APPROVED** [review]

**Pull request merged**